File: Math/2180. Count Integers With Even Digit Sum/2180. Count Integers With Even Digit Sum.py
class Solution:
    def countEven(self, num: int) -> int:
        def check_sum_even(num):
            summ = 0
            while num > 0:
                summ += num % 10
                num //= 10
            return summ % 2 == 0

        count = 0
        for n in range(1, num + 1):
            if check_sum_even(n):
                count += 1
        
        return count

# Time: O(n log n)
# Space: O(1)

File: Math/2180. Count Integers With Even Digit Sum/2180. Count Integers With Even Digit Sum.js
/**
 * @param {number} num
 * @return {number}
 */
var countEven = function (num) {
  const checkEvenSum = (num) => {
    let summ = 0;
    while (num > 0) {
      summ += num % 10;
      num = Math.floor(num / 10);
    }
    return summ % 2 === 0;
  };

  let count = 0;
  for (let i = 1; i <= num; i++) {
    if (checkEvenSum(i)) count++;
  }

  return count;
};


// Time: O(n log n)
// Space: O(1)